pkg/accounts/adaptor/repository/prisma.ts

```diff
@@ -0,0 +1,251 @@
+import { Option, Result } from '@mikuroxina/mini-fn';
+import { type PrismaClient } from '@prisma/client';
+
+import type { ID } from '../../../id/type.js';
+import type {
+  AccountFrozen,
+  AccountID,
+  AccountName,
+  AccountRole,
+  AccountSilenced,
+  AccountStatus,
+} from '../../model/account.js';
+import { Account } from '../../model/account.js';
+import type {
+  AccountRepository,
+  AccountVerifyTokenRepository,
+} from '../../model/repository.js';
+
+interface AccountPrismaArgs {
+  id: string;
+  name: string;
+  nickname: string;
+  mail: string;
+  passphrase_hash: string | null;
+  bio: string;
+  role: number;
+  frozen: number;
+  silenced: number;
+  status: number;
+  created_at: Date;
+  updated_at: Date | null;
+  deleted_at: Date | null;
+}
+
+export class PrismaAccountRepository implements AccountRepository {
+  readonly prisma: PrismaClient;
+
+  constructor(prisma: PrismaClient) {
+    this.prisma = prisma;
+  }
+
+  async create(account: Account): Promise<Result.Result<Error, void>> {
+    try {
+      await this.prisma.account.create({
+        data: {
+          ...this.toPrismaArgs(account),
+        },
+      });
+    } catch (e) {
+      return Result.err(e as Error);
+    }
+
+    return Result.ok(undefined);
+  }
+
+  async findByMail(mail: string): Promise<Option.Option<Account>> {
+    try {
+      const res = await this.prisma.account.findUnique({
+        where: {
+          mail: mail,
+        },
+      });
+      if (!res) {
+        return Option.none();
+      }
+      return Option.some(this.fromPrismaArgs(res));
+    } catch (e) {
+      return Option.none();
+    }
+  }
+
+  async findByName(name: string): Promise<Option.Option<Account>> {
+    try {
+      const res = await this.prisma.account.findUnique({
+        where: {
+          name: name,
+        },
+      });
+      if (!res) {
+        return Option.none();
+      }
+      return Option.some(this.fromPrismaArgs(res));
+    } catch (e) {
+      return Option.none();
+    }
+  }
+
+  private toPrismaArgs(account: Account): AccountPrismaArgs {
+    const role = ((role: AccountRole) => {
+      switch (role) {
+        case 'normal':
+          return 0;
+        case 'moderator':
+          return 1;
+        case 'admin':
+          return 2;
+      }
+    })(account.getRole);
```

Comment:
```suggestion
    const role = ({
      normal: 0,
      moderator: 1,
      admin: 2,
    } satisfies Record<AccountRole, number>)[account.getRole];
```
即時関数にしなくても, このようなオブジェクトによるマッピングで十分そうです.